### PR TITLE
Added new routing option: `SingleNodeRoutingInfo::RandomPrimary`

### PR DESCRIPTION
**Pull request by eifrah-aws**
- Added new routing option: `RandomPrimary`.
- Added new convenience method `Route::new_random_primary()`
- Added test cases to check correct routing for commands `EVAL`, `EVALSHA` and `FCALL` when no arguments are provided

This commit is (part of) the fix for this issue: https://github.com/valkey-io/valkey-glide/issues/1569

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: redis/src/aio/tokio.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use super::{AsyncStream, RedisResult, RedisRuntime, SocketAddr};
 use async_trait::async_trait;
+#[allow(unused_imports)] // silence warning in multiple configuration builds
```
Comment (shohamazon) [inline redis/src/aio/tokio.rs:3]: why is this necessary?

**Review by shohamazon — APPROVED** [review]
LGTM

File: redis/src/cluster.rs [hunk 2]
```diff
@@ -730,6 +727,9 @@ where
                         SingleNodeRoutingInfo::SpecificNode(route) => {
                             self.get_connection(&mut connections, route)?
                         }
+                        SingleNodeRoutingInfo::RandomPrimary => {
```
[the remark this answers is not in the document]
Comment (eifrah-aws) [reply redis/src/cluster.rs:730]: 1. The code for `RandomPrimary` is correct
2. The `RoutingInfo::SingleNode(SingleNodeRoutingInfo::Random)` was not changed (replaced the old code with a call to `Route::new_random_primary()`
 
Having said that, there is no bug on that line, however there can be an optimization: for example we could add a check to see whether the cmd is readonly, and in case it does, we could allow "read-from-replica" instead of "SlotAddr::Master`.

If you think that this is correct, we could open a new bug and fix it in a different PR (I prefer not to mix changes in a single PR)

File: redis/src/aio/tokio.rs (hunk 1 above)
Comment (eifrah-aws) [reply redis/src/aio/tokio.rs:3]: A relic from a previous change, will remove it (in my first commit, I also modified Cargo.toml)

File: redis/src/cluster_routing.rs [hunk 3]
```diff
@@ -949,6 +957,18 @@ impl Route {
...
+
+    /// Returns a new Route for a random primary node
+    pub fn new_random_primary() -> Self {
+        Self::new(random_slot(), SlotAddr::Master)
+    }
+}
+
+/// Choose a random slot from `0..SLOT_SIZE` (excluding)
+fn random_slot() -> u16 {
+    use rand::Rng;
```
Comment (ikolomi) [inline redis/src/cluster_routing.rs:969]: please put "use" alongside all the others at the top

**Review by ikolomi — APPROVED** [review]
